ingestor-default/src/test/scala/za/co/absa/hyperdrive/ingestor/implementation/utils/TestKafkaUtilDockerTest.scala

```diff
@@ -218,20 +200,22 @@ class TestKafkaUtilDockerTest extends FlatSpec with Matchers with BeforeAndAfter
     }
 
     withClue(){
-      records.size shouldBe messages.map(_._1).distinct.size
+      records.size shouldBe messages.map(r => (r._1, r._3)).distinct.size
     } withClue(s"This is likely an artifact of the test itself. You may want to increase waitForCompactionMillis." +
       s" The current value is $waitForCompactionMillis")
 
     val consumer = createConsumer(kafka)
     implicit val kafkaConsumerTimeout: Duration = kafkaSufficientTimeout
-    val actualRecords = KafkaUtil.getAtLeastNLatestRecordsFromPartition(consumer, new TopicPartition(topic, 0), 10)
+    val recordsPerPartition = (0 to partitions).map(p => new TopicPartition(topic, p) -> 4L).toMap
+    val actualRecords = KafkaUtil.getAtLeastNLatestRecordsFromPartition(consumer, recordsPerPartition)
     val values = actualRecords.map(_.value())
 
-    values.size should be >= 10
-    values should contain allElementsOf Seq("msg_100", "msg_99", "msg_98", "msg_96", "msg_94", "msg_92", "msg_90", "msg_88", "msg_86", "msg_84")
+    values.size should be >= 12
+    values should contain allElementsOf Seq("msg_103", "msg_102", "msg_101", "msg_100", "msg_99", "msg_97", "msg_95",
```

Comment:
This is really confusing which messages should stay in topic. 
If key is even or > 100 it should stay in topic, but what about "msg_97" .... i = 97, 97 % 2 != 0, i is not > 100 so it will have  key = 1  and it should not stay in topic

Reply:
99, 97, 95: Yes, they will have key=1, but they are the latest with key=1, so they stay as well.
But actually, I should check for 98, 96 and 94, while 84 is not necessary (it's the 13th item). Will fix it